### DISCUSSION TITLE
Split PDF Documents on UI

### DISCUSSION THREAD

**Discussion by mhupfauer** [jonaswinkler/paperless-ng#734] [Feature Requests]
## Description
A user can split PDFs documents within the `/documents/XXX` page based on the displayed page on the left hand side. If page 3 out of 25 is shown and the users clicks on e.g.: `Split` the current document would extract sites 4-25 to a new document and retain current properties for pages 1-3. The extracted document (pages 4-25) is fed into the consumption pipeline again.

## Use case
A user has a bulk quantity of documents that shall be scanned and does not want to restart the scan process for each individual correspondent. This feature will enable the user to put documents into the scanner, start the process and continue working on something else. After processing the user can split the document easily via the UI.

## Challenges (?)

1. Extract PDF current viewed site from embedded PDF reader?
2. Extract pages from .pdf file programmatic
3. Feed extracted document back into consumption pipeline?
4. Do not enable OCR / unpaper on initial bulk scan

**Comment by jonaswinkler** [comment]
Hello!

This is already a pinned issue (see #335) and something I want to work on next. I just did not have any time for this, and it's quite a big task.

Regarding your challenges:

1. Impossible with the iframe / embedded PDF reader. I simply ask the browser to display a PDF at a certain location on the page and everything beyond that is outside of my control.
2. This is easy, and the required dependencies are already in paperless.
3. Also easy. I just need to write an edited PDF document somewhere and call a function. Done.
4. Impossible, since I have no means to decide whether any given document is the result of a bulk scan or not. I certainly want to keep OCR turned on for non-bulk documents.

**Reply by mhupfauer** [reply]
Thanks for you reply!
Would it be possible to consume `bulk_XXXX.tiff` as bulk files and `scan_XXXX.tiff` as regular files. Most scanners have multiple options i.e. Scan to File, Scan to Image and so on. Those can be configured to name files differently (tested on Brother and Kyocera devices). The enduser than can be instructed to scan single documents as `Scan to File` and bulk documents as `Scan to Image`.

Maybe the consumer can be extended by a regex option to check for bulk or "regular" consumption?
